infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -616,13 +638,13 @@ private void onSinkClusterAddRemove(TopologyDescriptor newTopology) {
 
         // Only process new sinks if the local cluster role is SOURCE
         if (topologyDescriptor.getLocalClusterDescriptor().getRole() == ClusterRole.SOURCE) {
-            sessionManager.refresh(newTopology);
-            sessionManager.startReplication();
+            sessionManager.refresh(newTopology, fetchConnectionEndpoints());
```

Comment:
shouldn't connection endpoints come from newTopology?

Reply:
no, since connection endpoints depend on the type of cluster, it should come from the adapter implementation.
I havn't included it in the topologyDescriptor because it would not be used anywhere else.

Reply:
Changing the model as discussed offline.